src/nmodl/visitors/loop_unroll_visitor.cpp

```diff
@@ -28,6 +30,64 @@ static std::shared_ptr<ast::Expression> unwrap(const std::shared_ptr<ast::Expres
 }
 
 
+/**
+ * Replace any remaining instances of the FROM iteration variable with its value
+ */
+ast::StatementVector replace_iteration_var(const ast::StatementVector& statements,
+                                           const std::string_view index_var,
+                                           const int i) {
+    ast::StatementVector new_statements;
+    for (const auto& statement: statements) {
+        const auto& hits = collect_nodes(*statement, {ast::AstNodeType::VAR_NAME});
+        auto condition = [](const auto& hits, const auto& index_var) -> bool {
+            for (const auto& hit: hits) {
+                const auto& var = std::dynamic_pointer_cast<ast::VarName>(hit);
+                if (var->get_node_name() == index_var) {
+                    return true;
+                }
+            }
+            return false;
+        };
+        if (condition(hits, index_var)) {
+            /// The attentive reader may have an issue with using a regex
+            /// instead of something more sophisticated; here's an explanation
+            /// as to why we do it this way:
+            /// We know that our iteration variable, let's call it `i` for
+            /// concreteness, is of type VarName. However, the type which it
+            /// should be replaced by is either an Integer, or a Double.  The
+            /// "normal" way of doing this would be something like:
+            ///     const auto& parent = i->get_parent();
+            ///     const auto& new_var = std::make_shared<ast::Integer>("0", nullptr);
+            ///     new_var->set_parent(parent);
+            /// Sadly, this does not work, and about the only consistent
+            /// way of performing the replacement is by calling the
+            /// parents' `set_*` method.  The problem is that the type of
+            /// the parent can be virtually any class, and each class has
+            /// its own name and calling signature for the setter method.
```

Comment:
I do not agree. You do not have to write if statements and you do not have to write jinja stuff. You know the type of the child right? so, you need to call parent->set_integer or something similar. 

If you really need the type of the parent this function should be called from the parent itself. It checks its children and does the substitutions that you need to do. 

What does it mean that it can be virtually any class? Where can the from statement be? in what blocks? 

do the replacements in these visitors